src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -279,5 +300,427 @@ public async Task TestAwaitInLock_TopLevel()
         {
             await VerifyKeywordAsync("lock($$", LanguageVersion.CSharp9);
         }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnTask()
+        {
+            await VerifyKeywordAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    someTask.$$
+  }
+}
+", LanguageVersion.CSharp9, dotAwait: true, dotAwaitf: true);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnTaskOfT()
+        {
+            await VerifyKeywordAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task<int> someTask)
+  {
+    someTask.$$
+  }
+}
+", LanguageVersion.CSharp9, dotAwait: true, dotAwaitf: true);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnValueTask()
+        {
+            var valueTaskAssembly = typeof(ValueTask).Assembly.Location;
+            await VerifyItemExistsAsync(@$"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <MetadataReference>{valueTaskAssembly}</MetadataReference>
+        <Document FilePath=""Test2.cs"">
+using System.Threading.Tasks;
+
+class C
+{{
+  async Task F(ValueTask someTask)
+  {{
+    someTask.$$
+  }}
+}}
+        </Document>
+    </Project>
+</Workspace>
+", "await");
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnCustomAwaitable()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+public class DummyAwaiter: INotifyCompletion {
+    public bool IsCompleted => true;
+    public void OnCompleted(Action continuation) => continuation();
+    public void GetResult() {}
+}
+
+public class CustomAwaitable
+{
+    public DummyAwaiter GetAwaiter() => new DummyAwaiter();
+}
+
+static class Program
+{
+    static async Task Main()
+    {
+        var awaitable = new CustomAwaitable();
+        awaitable.$$;
+    }
+}", LanguageVersion.CSharp9, dotAwait: true);
```

Comment:
enh... i don't mind just always offering both.  it's ok for completion to have too many items.

Reply:
see https://github.com/dotnet/roslyn/pull/55817#discussion_r704280089